api_core/google/api_core/retry.py

```diff
@@ -156,9 +158,12 @@ def retry_target(target, predicate, sleep_generator, deadline, on_error=None):
         sleep_generator (Iterable[float]): An infinite iterator that determines
             how long to sleep between retries.
         deadline (float): How long to keep retrying the target.
-        on_error (Callable): A function to call while processing a retryable
-            exception.  Any error raised by this function will *not* be
-            caught.
+        on_error (Callable[Exception]): A function to call while processing a
+            retryable exception.  Any error raised by this function will *not*
+            be caught.
+        strict_deadline (bool): If :data:`True`, the last retry will run at
```

Comment:
What's the reason for a new argument to change the behavior versus using the "strict" logic all the time?

Reply:
Backward compatibility in order to not break any existing behavior of any of the libraries.

It might be unnecessary, but played it safe here. I can still make this _the_ behavior, though, if there are no objections from anyone?

Reply:
I _think_ it won't break anything, but definitely want @busunkim96 and @crwilcox's thoughts.

Reply:
On second thought, if we make timeout per-request to match the semantics of the GAPIC clients, this whole change is unnecessary.

That does mean someone needs to configure both the retry arg and the timeout arg to set a total timeout, but I think consistency with GAPIC is more important than the usability improvement of having a multi-request timeout option.